commit 611fbeb44a777e5ab54ab3127ec85f72147911d8
Author: Al Viro <viro@zeniv.linux.org.uk>
Date:   Thu Aug 22 05:39:32 2024 +0100

    selftests:core: test coverage for dup_fd() failure handling in unshare_fd()
    
    At some point there'd been a dumb braino during the dup_fd()
    calling conventions change; caught by smatch and immediately fixed.
    The trouble is, there had been no test coverage for the dup_fd() failure
    handling - neither in kselftests nor in LTP.  Fortunately, it can be
    triggered on stock kernel - ENOMEM would require fault injection, but
    EMFILE can be had with sysctl alone (fs.nr_open).
    
    Add a test for dup_fd() failure.
    Fixed up commit log and short log - Shuah Khan
    
    Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
    Signed-off-by: Shuah Khan <skhan@linuxfoundation.org>

diff --git a/tools/testing/selftests/core/Makefile b/tools/testing/selftests/core/Makefile
index ce262d097269..8e99f87f5d7c 100644
--- a/tools/testing/selftests/core/Makefile
+++ b/tools/testing/selftests/core/Makefile
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0-only
 CFLAGS += -g $(KHDR_INCLUDES)
 
-TEST_GEN_PROGS := close_range_test
+TEST_GEN_PROGS := close_range_test unshare_test
 
 include ../lib.mk